### PR TITLE
LPS-132437 Migrate dynamic data template selection modals

### PR DESCRIPTION
**Pull request by markocikos**
Hey @liferay-data-engine ,

This is a part of [an ongoing epic](https://issues.liferay.com/browse/LPS-129246) to replace legacy AUI-based modals with Clay/React-based `openSelectionModal`.  We have previously migrated DDM modals, in https://github.com/liferay-data-engine/liferay-portal/pull/407, but we were not aware of `openDDMPortlet` utilily. This PR is first of two tasks to migrate and deprecate this utility.

JIRA: https://issues.liferay.com/browse/LPS-132437

Notes:
- There are 5 uses of `openDDMPortlet`, 4 of which is a selection modal. One, in this PR, is not, and we are replacing it with `openModal`. This way we can cleanly deprecate the util at the end of the epic. 
- In this PR, we are slightly changing the appearance of modal in DDM taglib (asset publisher config). A custom fixed width (1024px) is not standard by [Lexicon](https://liferay.design/lexicon/core-components/modals/#size), instead we are choosing one of standard `size`s. For a complex modal content such as management toolbar and search container, it is standard to use a full screen modal, so I set it that way in this PR. Please see before and after gifs.
- As usual, we are taking the opportunity to modernize related code, and remove AUI.

Test case 1:
1. Content & Data > Kaleo Forms Admin > Add a new process
1. Follow workflow to step 4.
1. Task > (⋮) > Assign Form
1. Create a form if there are none
1. Select a form

After:
after1

Test case 2:
1. Place Asset Publisher app in a page.
1. (⋮) > Configuration > Display Settings > Manage Templates
1. Create a new template.
1. Close modal
1. Confirm newly created template is available in dropdown

Before:
before

After:
after

/cc @kresimir-coko @jonmak08

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/resources/META-INF/resources/admin/process/task_template_search_container.jsp [hunk 1]
```diff
@@ -107,61 +107,50 @@ KaleoTaskFormPair initialStateKaleoTaskFormPair = KaleoFormsUtil.getInitialState
...
+			"backURL", HtmlUtil.escapeURL(backURL)
+		).put(
+			"itemSelectorURL", itemSelectorURL
+		).put(
+			"portletNamespace", liferayPortletResponse.getNamespace()
+		).put(
+			"saveInPortletSessionURL", saveInPortletSessionURL
+		).build()
+	%>'
+	module="admin/js/KaleoFormsTemplateSelector"
```
Comment (diegonvs) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-forms-web/src/main/resources/META-INF/resources/admin/process/task_template_search_container.jsp:150]: ✨

**Comment by choliveira84** [comment]
Hey @markocikos

Could check the status of relevant test? Apparently, it's related to the JSP.

Let me know if you need any help on this.

Thanks.

**Comment by markocikos** [comment]
@choliveira84 I force-pushed with the dependency fix in https://github.com/liferay-data-engine/liferay-portal/pull/454/commits/d2e0c773b1bde124296f1754dc6bd18c713593fd. I am not sure how it's possible that everything worked locally without this :thinking: It took a `gradle clean deploy` to replicate the issue. I retested after the compile fix, all seems to be good. Let's see what CI has to say...

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
Hi, @markocikos CI seems stuck. Could you please close this PR and send us a new one with your recent changes?

Thanks and sorry for the inconvenience.

**Comment by rodrigopaulino** [comment]
Resent https://github.com/liferay-data-engine/liferay-portal/pull/466

**Pull request closed without merging**